src/pydvl/utils/utility.py

```diff
@@ -134,8 +135,8 @@ def __init__(
         score_range: Tuple[float, float] = (-np.inf, np.inf),
         catch_errors: bool = True,
         show_warnings: bool = False,
-        enable_cache: bool = False,
-        cache_options: Optional[MemcachedConfig] = None,
+        cache_backend: Optional[CacheBackend] = None,
+        cached_func_options: CachedFuncConfig = CachedFuncConfig(),
```

Comment:
Default value is mutable. I would either freeze the dataclass in the declaration or use None and a check below